configure.ac

```diff
@@ -2545,77 +2545,90 @@ fi
 # }}}
 
 # --with-libdpdk {{{
+with_libdpdk_cppflags=""
+with_libdpdk_ldflags=""
 AC_ARG_WITH(libdpdk, [AS_HELP_STRING([--with-libdpdk@<:@=PREFIX@:>@], [Path to the DPDK build directory.])],
 [
 	if test "x$withval" != "xno" && test "x$withval" != "xyes"
 	then
-		RTE_BUILD="$withval"
+		with_libdpdk_cppflags="-I$withval/include"
+		with_libdpdk_ldflags="-L$withval/lib"
 		with_libdpdk="yes"
 	else
-		RTE_BUILD="/usr"
-		with_libdpdk="$withval"
+		if test "x$withval" = "xno"
+		then
+			with_libdpdk="no (disabled)"
+		fi
 	fi
-	DPDK_INCLUDE="$RTE_BUILD/include"
-	DPDK_LIB_DIR="$RTE_BUILD/lib"
-	FOUND_DPDK=yes
 ], [with_libdpdk="no"])
 
 if test "x$with_libdpdk" = "xyes"
 then
-	LOCAL_DPDK_INSTALL="no"
-	AC_CHECK_HEADER([$DPDK_INCLUDE/rte_config.h], [LOCAL_DPDK_INSTALL=yes],
-		[AC_CHECK_HEADER([$DPDK_INCLUDE/dpdk/rte_config.h],
-		[],
-		[FOUND_DPDK=no], [])], [])
-
-	if test "x$LOCAL_DPDK_INSTALL" = "xno"
-	then
-		DPDK_INCLUDE=$DPDK_INCLUDE/dpdk
-	fi
+	SAVE_CPPFLAGS="$CPPFLAGS"
+	CPPFLAGS="$CPPFLAGS $with_libdpdk_cppflags"
+	AC_CHECK_HEADERS(rte_config.h, [with_libdpdk="yes"],
+		[
+			AC_CHECK_HEADERS(dpdk/rte_config.h,
+				[
+					with_libdpdk="yes"
+					with_libdpdk_cppflags="$with_libdpdk_cppflags/dpdk"
```

Comment:
I think it's disgusting ;)
Seriously, what problem are you trying to solve?

> If rte_xxx.h includes others rte_xxx.h (e.g.: using just #include <rte_xxx.h>) the compilation will fail.

Then either those headers are buggy, or we're including from the wrong level.

> Dpdk headers can be installed directly to include dir or to dpdk sub-folder.

I checked the dpdk-devel package on Fedora, and it has the headers in /usr/include/dpkd/
So we should add /usr/include/dpkd to the include path and check for rte_config.h instead of checking for dpdk/rte_config.h. If I a user wants to build dpdk from source, then we should allow him to specify the include path, and check for the same header there. Alternatively allow him to override CPPFLAGS and make it the users problem. Presumably if you can build dpdk from source, you can figure out what CPPFLAGS you need. We shouldn't add hacks to configure.ac to cater to this use case.